src/session.rs

```diff
@@ -244,8 +258,34 @@ impl Session {
             .unwrap();
     }
 
-    // Receives and dispatches frames from the connection to the corresponding
-    // channels.
+    /// Executes the heartbeat routine for the given connection.
+    ///
+    /// Calling this function should not take place on the main thread. Will block indefinitely.
+    fn heartbeat_loop(mut connection: Connection, heartbeat_interval: u16) -> () {
+        debug!("Starting heartbeat loop.");
+        let sleep_interval = Duration::from_secs(heartbeat_interval as u64);
+        let heartbeat_frame = Frame {
+            frame_type: FrameType::HEARTBEAT,
+            channel: 0,
+            payload: FramePayload::new(vec![]),
+        };
+
+        loop {
+            // Emit a new heartbeat frame & panic if any failure occurred.
+            let res = connection.write(heartbeat_frame.clone());
+            if let Err(err) = res {
+                connection.shutdown();
```

Comment:
The result of that function call is being discarded. Can we do this explicit instead of implicit?